python/src/deltachat/testplugin.py

```diff
@@ -224,6 +226,13 @@ def finalize(self):
                 acc.shutdown()
                 acc.disable_logging()
 
+        def new_imap_conn(self, account, config_folder=None):
+            imap_conn = ImapConn(account)
+            self._finalizers.append(imap_conn.shutdown)
```

Comment:
I liked the solution with `def __del__(self):` better because someone could forget to call `self._finalizers.append(imap_conn.shutdown)` (or use ImapConn without Pytest) while __del__() will always be called.

Reply:
using __del__ is very difficult business.  Should be avoided if possible. 
also we are in a test run and have shutdown machinery so no need to try to work automatically.